build/package/scripts/build-java.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+set -eu
+
+OUTPUT_DIR="docker"
+WORKING_DIR="."
+ROOT_DIR=$(pwd)
+ARTIFACTS_DIR=$ROOT_DIR/.ods/artifacts
+ARTIFACT_PREFIX=
+DEBUG="false"
+GRADLE_ADDITIONAL_TASKS=
+GRADLE_OPTIONS=
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+
+    --working-dir) WORKING_DIR="$2"; shift;;
+    --working-dir=*) WORKING_DIR="${1#*=}";;
+
+    --output-dir) OUTPUT_DIR="$2"; shift;;
+    --output-dir=*) OUTPUT_DIR="${1#*=}";;
+
+    --gradle-additional-tasks) GRADLE_ADDITIONAL_TASKS="$2"; shift;;
+    --gradle-additional-tasks=*) GRADLE_ADDITIONAL_TASKS="${1#*=}";;
+
+    # Gradle project properties ref: https://docs.gradle.org/current/userguide/build_environment.html#sec:gradle_configuration_properties
+    # Gradle options ref: https://docs.gradle.org/current/userguide/command_line_interface.html
+    --gradle-options) GRADLE_OPTIONS="$2"; shift;;
+    --gradle-options=*) GRADLE_OPTIONS="${1#*=}";;
+
+  *) echo "Unknown parameter passed: $1"; exit 1;;
+esac; shift; done
+
+if [ -z "${WORKING_DIR}" ]; then
+  WORKING_DIR="${ROOT_DIR}"
+fi
+
+if [ "${DEBUG}" == "true" ]; then
+  set -x
+  echo
+  echo "Debug mode enabled (DEBUG=true)"
+  echo "Generating debug logs ..."
+  echo "... current folder:"
+  pwd
+  echo "... folder content:"
+  ls -lart
+  echo
```

Comment:
I find this overly wordy. Once you do `set -x`, each line is printed before it is executed. On line 49 you see into which dir you change, which gives removes the need for the `pwd` here.